modules/base/rendering/renderablesphereimagelocal.cpp

```diff
@@ -70,24 +65,24 @@ RenderableSphereImageLocal::RenderableSphereImageLocal(
     const Parameters p = codegen::bake<Parameters>(dictionary);
 
     _texturePath = p.texture.string();
-    _textureFile = std::make_unique<ghoul::filesystem::File>(_texturePath.value());
-    _texturePath.onChange([this]() {
-        loadTexture();
-    });
+    _texturePath.onChange([this]() { _texture->loadFromFile(_texturePath.value()); });
     addProperty(_texturePath);
-    _textureFile->setCallback([this]() { _textureIsDirty = true; });
 }
 
 bool RenderableSphereImageLocal::isReady() const {
     return RenderableSphere::isReady() && _texture;
 }
 
+void RenderableSphereImageLocal::initialize() {
+    _texture = std::make_unique<TextureComponent>(2);
```

Comment:
Can we do all of this in the `initialize` or does it have to go into the `initializeGL`?